### PR TITLE
fix(tutorial): Update ping tutorial to match examples/ping.rs

### PR DESCRIPTION
**Pull request by wizeguyy**
## Description
While following the ping tutorial myself, I noticed it was out of date from the ping example in `examples/ping.rs`, and the code given in the tutorial no longer cleanly builds. I decided to update the tutorial to build the exact code in the example.

## Change checklist
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] ~I have added tests that prove my fix is effective or that my feature works~ (not applicable)
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by wizeguyy** [comment]
I'm not sure if this warrants a change log entry, and if so, where I should add that. Please advise.

File: src/tutorials/ping.rs [hunk 1]
```diff
@@ -136,32 +136,37 @@
 //! [plaintext](crate::plaintext). It only cares about _what_ messages are sent
 //! on the network.
 //!
+//!
```
Comment (mxinden) [inline src/tutorials/ping.rs]: Why is this newline needed?

File: src/tutorials/ping.rs [hunk 2]
```diff
@@ -136,32 +136,37 @@
...
 //! The two traits [`Transport`] and [`NetworkBehaviour`] allow us to cleanly
 //! separate _how_ to send bytes from _what_ bytes to send.
 //!
 //! With the above in mind, let's extend our example, creating a [`ping::Behaviour`](crate::ping::Behaviour) at the end:
```
Comment (mxinden) [inline src/tutorials/ping.rs:142]: With the change below, "creating" is no longer accurate, i.e. the code below is not actually instantiating a `Behaviour`. Can you either update the comment above, or actually instantiate the `Behaviour` in the code below?

``` rust
let behaviour = Behaviour::default();

Ok(())
```

File: src/tutorials/ping.rs [hunk 3]
```diff
@@ -174,28 +179,32 @@
...
-//!
-//!     let mut swarm = Swarm::with_async_std_executor(transport, behaviour, local_peer_id);
+//!     let mut swarm = Swarm::with_async_std_executor(transport, Behaviour::default(), local_peer_id);
 //!
 //!     Ok(())
 //! }
+//!
+//! /// Our network behaviour.
+//! ///
+//! /// For illustrative purposes, this includes the [`KeepAlive`](behaviour::KeepAlive) behaviour so a continuous sequence of
```
Comment (mxinden) [inline src/tutorials/ping.rs]: Needs to be wrapped at 80 chars.

**Review by mxinden — COMMENTED** [review]
Thanks a lot for being proactive here and providing a patch @wizeguyy. Mind addressing the comments below? Otherwise this looks good to me.

File: src/tutorials/ping.rs (hunk 1 above)
Comment (wizeguyy) [reply src/tutorials/ping.rs]: Oops, that was by accident. Will remove it.

File: src/tutorials/ping.rs (hunk 2 above)
Comment (wizeguyy) [reply src/tutorials/ping.rs:142]: Good point. Will fix.

File: src/tutorials/ping.rs (hunk 3 above)
Comment (wizeguyy) [reply src/tutorials/ping.rs]: Agreed. Will fix.

Comment (wizeguyy) [reply src/tutorials/ping.rs]: On this subject, have you considered adding a `rustfmt.toml` to the project, to automate and standardize style items like this?

**Comment by wizeguyy** [comment]
> Thanks a lot for being proactive here and providing a patch @wizeguyy. Mind addressing the comments below? Otherwise this looks good to me.

Thanks for the feedback! I've addressed your comments and amended the original commits.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Formatting job fails, otherwise LGTM

**Comment by wizeguyy** [comment]
> Formatting job fails, otherwise LGTM

@thomaseizinger fixed. Sorry, I should've caught that sooner...

File: src/tutorials/ping.rs [hunk 4]
```diff
@@ -136,32 +136,39 @@
 //! [plaintext](crate::plaintext). It only cares about _what_ messages are sent
 //! on the network.
 //!
+//!
```
Comment (thomaseizinger) [inline src/tutorials/ping.rs]:
```suggestion
```

**Review by thomaseizinger — APPROVED** [review]

File: src/tutorials/ping.rs (hunk 4 above)
Comment (thomaseizinger) [reply src/tutorials/ping.rs]: Can you apply these changes? I don't seem to have permission.

Comment (wizeguyy) [reply src/tutorials/ping.rs]: Done.

Comment (wizeguyy) [reply src/tutorials/ping.rs]: Let me squash it in to the original

Comment (wizeguyy) [reply src/tutorials/ping.rs]: Ok, squashed.

**Review by thomaseizinger — DISMISSED** [review]

**Comment by thomaseizinger** [comment]
Your PR doesn't seem to allow us to update with the base branch and thus our merge automation is stuck. See https://github.com/libp2p/rust-libp2p/pull/3289/checks?check_run_id=10766359069. Can you try and fix this please? :)

**Comment by wizeguyy** [comment]
> Your PR doesn't seem to allow us to update with the base branch and thus our merge automation is stuck. See https://github.com/libp2p/rust-libp2p/pull/3289/checks?check_run_id=10766359069. Can you try and fix this please? :)

I've never seen this before... I typically disallow maintainer edits, as it invalidates PGP signatures. Never had it get in the way like this before.

I've removed the restriction, so your pipeline should be able to edit my commits as needed now.

**Review by thomaseizinger — DISMISSED** [review]

**Comment by wizeguyy** [comment]
I see one of the CI tests failed, but I am unable to reproduce or diagnose that failure on my own. If anyone can provide a bit of explanation what failed (or show me how to understand the test results myself), that would be helpful.

Funny how sometimes the smallest PRs become such a large hassle :sweat_smile:

**Comment by thomaseizinger** [comment]
> I see one of the CI tests failed, but I am unable to reproduce or diagnose that failure on my own.

Not your fault, we recently added these interop tests and they are not fully stable yet!

Sorry that it hasn't been as smooth. Leave it with us, it will merge eventually :)

(There is a subscription setting to only be notified on merges in case you want to avoid the spam.)

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Review by mxinden — APPROVED** [review]

**Pull request merged**